File: src/main/java/com/mudhut/nudge/users/services/helpers/PasswordValidator.java
package com.mudhut.nudge.users.services.helpers;

import org.springframework.stereotype.Component;

@Component
public class PasswordValidator {

    public void validatePassword(String password) {
        if (password == null || password.length() < 8) {
            throw new IllegalArgumentException("Password must be at least 8 characters long");
        }

        boolean hasLetter = false;
        boolean hasDigit = false;
        boolean hasSpecial = false;

        for (char c : password.toCharArray()) {
            if (Character.isLetter(c)) {
                hasLetter = true;
            } else if (Character.isDigit(c)) {
                hasDigit = true;
            } else if (!Character.isWhitespace(c)) {
                hasSpecial = true;
            }
        }

        if (!hasLetter || !hasDigit || !hasSpecial) {
            throw new IllegalArgumentException(
                    "Password must contain at least one letter, one number, and one special character");
        }
    }
}

File: src/main/kotlin/com/mudhut/nudge/users/services/helpers/PasswordValidator.kt
package com.mudhut.nudge.users.services.helpers

import org.springframework.stereotype.Component

@Component
class PasswordValidator {

    fun validatePassword(password: String) {
        if (password.length < 8) {
            throw IllegalArgumentException("Password must be at least 8 characters long")
        }

        var hasLetter = false
        var hasDigit = false
        var hasSpecial = false

        for (c in password) {
            when {
                c.isLetter() -> hasLetter = true
                c.isDigit() -> hasDigit = true
                !c.isWhitespace() -> hasSpecial = true
            }
        }

        if (!hasLetter || !hasDigit || !hasSpecial) {
            throw IllegalArgumentException(
                "Password must contain at least one letter, one number, and one special character"
            )
        }
    }
}
